grid/docs-cell-type-renderers (#23178)
highcharts/highcharts: docs/grid/cell-content.md

### Text input
Renders an editable text field for the value. It can also render static HTML elements. This is the default renderer for most cases.

```js
{
  id: 'username', // column id
  dataType: 'string',
  cells: {
    renderer: {
      type: 'input'
    },
    editable: true
  }
}
```

### Checkbox input
Renders a native checkbox input element.

```js
{
  id: 'active', // column id
  dataType: 'boolean',
  cells: {
    renderer: {
      type: 'checkbox'
    }
  }
}
```

### Date input
Renders a native date input that supports HTML datepicker.

```js
{
  id: 'date_date', // column id
  dataType: 'datetime',
  cells: {
    renderer: {
        type: 'dateInput'
    }
  }
}
```

### Select
Renders a dropdown select menu for predefined options.

```js
{
  id: 'country', // column id
  dataType: 'string',
  cells: {
    renderer: {
      type: 'select',
      options: [
        { value: 'NO', label: 'Norway' },
        { value: 'NL', label: 'Netherlands' },
        { value: 'PL', label: 'Poland' },
        { value: 'EC', label: 'Ecuador' }
      ]
    }
  }
}
```

### Sparkline
Renders an inline miniature chart (e.g. bar, line) inside a cell using Highcharts.

> You can configure chart by the `chartOptions` API option, that supports all Highcharts options.

Reviewer: ```suggestion
You can configure chart by the `chartOptions` API option, that supports all Highcharts configurations.
```
I think it 'flows' better to no use the word 'option' twice so close together

Note: Customizing cell content is not part of Highcharts Grid Lite, so refer to [install instructions](https://www.highcharts.com/docs/dashboards/grid-standalone) for the full version to enable this functionality.

Highcharts Grid supports different cell renderers to provide interactive data presentation inside table cells. You can define the renderer property for each column in the `columns[].cells` configuration object to specify how each cell should be displayed or interacted with.

Renderers can be used to control how values are displayed `cells.renderer` and how they behave in edit mode `cells.editMode.renderer`. This allows you to, for example, show plain text by default and present a checkbox or input field when a cell becomes editable.

Supported renderers:

> * text input
> * checkbox
> * date
> * select
> * sparkline

Reviewer 2: ```suggestion
* `'text'`
* `'textInput'`
* `'dateInput'`
* `'checkbox'`
* `'select'`
* `'sparkline'`
```

Author: Table is more attractive

## Renderers types

> In the [renderer](https://api.highcharts.com/dashboards/#interfaces/Grid_Options.ColumnOptions#renderer) API option you can customize content of the cell, based on dataType.

Reviewer 2: ```suggestion
In the [renderer](https://api.highcharts.com/dashboards/#interfaces/Grid_Options.ColumnOptions#renderer) API option, you can set the default cell renderer for view and edit modes. If not specified, it is determined by the `dataType` option.
```

### Text input
Renders an editable text field for value.

```js
{
  id: 'username', // column id
  dataType: 'string',
  cells: {
    renderer: {
      type: 'input'
    },
    editable: true
  }
}
```

### Checkbox input
Renders a native checkbox input element.

```js
{
  id: 'active', // column id
  dataType: 'boolean',
  cells: {
    renderer: {
      type: 'checkbox'
    }
  }
}
```

### Date input
Renders a native date input that supports HTML datepicker.

```js
{
  id: 'date_date', // column id
  dataType: 'datetime',
  cells: {

>     format: '{value:%Y-%m-%d}',

Reviewer 2: ```suggestion
```

    renderer: {
        type: 'dateInput'
    }
  }
}
```

### Select
Renders a dropdown select menu for predefined options.

```js
{
  id: 'country', // column id
  dataType: 'string',
  cells: {
    renderer: {
      type: 'select',
      options: [
        { value: 'NO', label: 'Norway' },
        { value: 'NL', label: 'Netherlands' },
        { value: 'PL', label: 'Poland' },
        { value: 'EC', label: 'Ecuador' }
      ]
    }
  }
}
```

### Sparkline
Renders an inline miniature chart (e.g. bar, line) inside a cell using Highcharts.

You can configure chart by the `chartOptions` API option, that supports all Highcharts options.

```js
{
  id: 'trend', // column id
  cells: {
    renderer: {
      type: 'sparkline',
      chartOptions: {
        chart: {
          type: 'bar'
        },
        plotOptions: {
          series: {
            dataLabels: {
              enabled: true
            },
            negativeColor: "#f00"
          }
        }
      }
    }
  }
}
```

> Please note that you should include the `highcharts.js` file.

Reviewer 2: ```suggestion
Please note that you should include the `highcharts.js` file before including the Grid library. If you do it the other way around, or use ES Modules, you should connect Highcharts manually using: `Grid.CellRendererRegistry.types.sparkline.useHighcharts(Highcharts);`
```
